vrs/marten-construction-script.py

```diff
@@ -25,156 +42,323 @@
+    ht_minrep = hl.split_multi(ht_minrep)
+    ht_minrep = ht_minrep.filter(
+        ht_minrep.alleles[1] != ht_minrep.original_alleles[ht_minrep.a_index]
+    )
+
+    # 5 partitions to yield 3000 min-rep variants, the default
+    minrep_partitions = 5
+    if args.custom:
+        minrep_partitions = int(counts_list[-1]) // 600
+        if minrep_partitions < 1 and counts_list[-1] != 0:
+            minrep_partitions = 1
+    logger.info(f"Partitions to search for min-rep: %i", minrep_partitions)
+
+    ht_minrep = ht_minrep._filter_partitions(
+        sample(range(ht_minrep.n_partitions()), minrep_partitions)
+    )
+    ht_minrep = ht_minrep.select()
+    ht_minrep = ht_minrep.join(ht_whole, how="inner")
+    # an inner join here joins on Locus and Allele
+    # doing the same using a collect() and filter() step would be awkward, since it involves two separate fields
+
+    # checkpoint for final minrep variants
+    ht_minrep = ht_minrep.checkpoint(
+        f"{args.tmp_dir}/minrep-checkpoint-v2-checking.ht",
+        _read_if_exists=args.read_if_exists,
+        overwrite=args.overwrite,
+    )
+
+    # Counts and further assembly
+    logger.info(f"There are %i variants in the minrep set", ht_minrep.count())
+    ht_final = ht_union.union(ht_multiallelic)
+    ht_final = ht_final.union(ht_minrep)
+
+    # Check for duplicates and de-duplication
+    org_count = ht_final.count()
+    ht_final = ht_final.distinct()
+    duplicates_removed = org_count - ht_final.count()
+    logger.info(f"Number of duplicates removed: %i", duplicates_removed)
+
+    # Export final Table and VCF (both zipped and unzipped) and append header info for missing FILTER descriptions
+    logger.info(f"Before exporting, there are %i total variants", ht_final.count())
+    ht_final = ht_final.checkpoint(
+        f"{args.final_path}/testing-vcf-export.ht",
+        overwrite=args.overwrite,
+        _read_if_exists=args.read_if_exists,
+    )
+    logger.info("VCF as Hail Table written to final path")
+    if args.export_vcf:
+        hl.export_vcf(
+            ht_final,
+            f"{args.final_path}/testing-vcf-export.vcf",
+            append_to_header=args.header_fix_path,
+        )
+        logger.info("VCF as Unzipped VCF written to final path")
+    if args.export_bgz:
+        hl.export_vcf(
+            ht_final,
+            f"{args.final_path}/testing-vcf-export.vcf.bgz",
+            append_to_header=args.header_fix_path,
+        )
+        logger.info("VCF as Zipped VCF.BGZ written to final path")
+
+    logger.info("Script and output is finished!")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--tmp-dir",
+        help="Path for storing checkpoints, default to gs://gnomad-tmp-4day/vrs",
+        default=TMP_DIR,
+    )
+    parser.add_argument(
+        "--py_rand_seed", help="Random seed for python, def: 505", default=505, type=int
```

Comment:
OH okay noted! I had been naming them with underscores with "--py_rand_seed" because in the team's scripts I had seen all of the arguments called from "args.name_exapmle" with underscores in their name . Will fix!